### DISCUSSION TITLE
Warp image with "wheel color" (aka. flow)

### DISCUSSION THREAD

**Discussion by b4zz4** [ImageMagick/ImageMagick#2927] [Digital Image Processing]
I have a color degrade and want to distort it using an image that represents the color shift and saturation intensity.

Example:

![map](https://user-images.githubusercontent.com/643833/101187562-45f75080-3633-11eb-9d5a-a97f832f37c6.png)
+ 
![a](https://user-images.githubusercontent.com/643833/101187575-4b549b00-3633-11eb-9182-f5ec8c04d6b0.png)
### result:  
![map1](https://user-images.githubusercontent.com/643833/101187597-53143f80-3633-11eb-89e3-f01ad97ffc50.png)

For this I use a neural network, but I think it can be built with imagick.

**Comment by fmw42** [comment]
ImageMagick can distort that using the first two channels of the second image as follows:

`convert red_green.png hand.png -define compose:args=200,200 -compose displace -composite x.png`

![x](https://user-images.githubusercontent.com/13458218/101193556-965db680-3611-11eb-9c1b-b33af19d0ff8.png)

So here the red and green channels of the hand image are used to distort in the x and y directions respectively.  You can try swapping channels to use different channels to control the distortion.

**Comment by snibgo** [comment]
The question is not clear.

> I have a color degrade and want to distort it ...

Do you want to distort the colours, or a geometric distortion (which moves pixels)?

You show three images. Are you asking how to make the third image from the first two, usng ImageMagick?

**Comment by b4zz4** [comment]
@snibgo 
I do not know if it is possible, it would be a polar displacement depending on the color (HUE) and the saturation as displacement intensity.

![polar color](https://raw.githubusercontent.com/alexanderb14/PatchMatch-Optical-Flow/master/results/colors.png)

**Comment by b4zz4** [comment]
## RED
convert hand.png -colorspace HSL -channel H -separate red.png

## GREEN
convert hand.png -modulate 100,100,-130 -colorspace HSL -channel H -separate green.png

## DE-SATURATION
convert hand.png -colorspace HSL -channel L -separate - | convert - -negate -modulate 200 l.png

## BLUE 
convert -size $(convert hand.png info:- | cut -d" " -f3) xc:black blue.png

## MIX
convert red.png green.png blue.png  -combine -set colorspace RGB  - | convert - l.png -compose multiply -composite rb_wheel.png

## DISPLACE
convert red_green.png rb_wheel.png -compose Difference -composite salida1.png

## My approximate

![salida1](https://user-images.githubusercontent.com/643833/101209643-d98c4980-3652-11eb-89f9-1286d89d2dca.png)
